app/models/child_object.rb

```diff
@@ -22,8 +22,14 @@ def check_for_size_and_file
     width_and_height(remote_metadata)
   end
 
-  def processing_event(message, status = 'info', current_batch_process = parent_object&.current_batch_process, _current_batch_connection = parent_object&.current_batch_connection)
-    IngestNotification.with(parent_object_id: parent_object&.id, child_object_id: id, status: status, reason: message, batch_process_id: current_batch_process&.id).deliver(User.first)
+  def processing_event(message, status = 'info', _current_batch_process = parent_object&.current_batch_process, current_batch_connection = parent_object&.current_batch_connection)
+    return unless current_batch_connection
+    ie = IngestEvent.new(
```

Comment:
just do create! instead of new and then save!, that way you do not alloc and then free a variable from memory